udata/search/fields.py

```diff
@@ -56,7 +58,20 @@ def get_value_filter(self, value):
 
 
 class TermsFacet(Facet, DSLTermsFacet):
-    pass
+
+    def add_filter(self, filter_values):
+        if not filter_values:
```

Comment:
It doesn't coerce to a list, `all` perform a boolean product (ie. apply `AND` on all members) and `any same with `OR`.
I made (at least) 2 mistakes in my sample code (I wrote it without testing it):
- `all` and `any` works in shortcircuit evalutation (ie. stops whenever it can so doesn't evaluate all proposition).
- both expect a list or an iterator, not variables args

I'll make a PR to your PR with a tested proposal